### ISSUE TITLE
Website is Ugly

### ISSUE THREAD

**Issue by PullJosh** [ElementalCode/elemental#220]
Sorry to whoever designed it. :package:

I hate to break it to all y'alls, but the website looks like it was made by kids (which, admittedly, it was). I feel like [my first design](http://codepen.io/PullJosh/pen/JdQMgq) looked _pretty_ nice, but I can do better. The other issue is that it didn't really give all of the information we need to give. :P

Before the public beta, I want to overhaul the frontend of the website (hopefully that's okay!) Before I can make anything productive, though, I need to know exactly what information each page needs to communicate. I don't know if we are really to that point yet. If we are, let me know what needs to happen. If not, let me know as soon as we have it figured out.

Hopefully you guys agree!

**To-Do List:**
- [ ] Make cta buttons align with width of header image
- [x] Hover effect for navigation links
- [ ] Some sort of "branding" image where green Elemental link currently is in navigation
- [ ] Add "carousels"
- [ ] Pop-up sign in box
- [ ] Page footer

**Comment by matthewr6** [comment]
Any pages in specific you want to change, or just everything?
Also any specific things about the design that you don't like?  I personally like it (though having no border-radius might be better, and with less shadow)

Also, two things.
First - NO DARK BACKGROUNDS please
Second - as we said about that design, remember that none of the pages are simply static cover pages

**Comment by PullJosh** [comment]
Everything.

1) That's fine
2) Yup! That's why I need everyone to let me know what content goes where.

**Comment by matthewr6** [comment]
For 2, basically see the pages as they stand right now - they have all the information that they'll probably have (although make sure to make it flexible w/adding or removing sections/info)

**Comment by PullJosh** [comment]
Okay! I'm assuming that w/ python it's relatively easy to insert content which follows a specific format?
(eg. if we have carousels like on the scratch homepage, you can fill in images & text where I just put placeholders)

**Comment by matthewr6** [comment]
Yeah.

Also

> Also any specific things about the design that you don't like? I personally like it (though having no border-radius might be better, and with less shadow)

**Comment by BookOwl** [comment]
I like what we have now.

**Comment by PullJosh** [comment]
> I like what we have now.

I dislike what we have right now so much that I'm going to make the executive decision that we need a new design. I know this is a group project, but I can't stand what we've got. :package:

**Comment by matthewr6** [comment]
I'm going to veto that because you need to at least explain why first and propose actual changes instead of just saying you'll redo it because you personally dislike it without explaining anything else

**Comment by BookOwl** [comment]
What exactly is wrong with what we have now?

**Comment by matthewr6** [comment]
That's been asked three times so far :P

**Comment by BookOwl** [comment]
> the website looks like it was made by kids

What makes you say that?

**Comment by PullJosh** [comment]
Compare [our website](https://elementalcode.herokuapp.com/) to something more professional like Scratch. We have massive border-radiuses (radii?) and box-shadows, both of which are very 2004. The navigation bar, on hover, is a terrible off-green, and there is a bunch of unused space.

Take a look at the site from the eyes of a potential user. Nothing about it appears professional. Even on a subconscious level, the site as it stands will turn away users.
...
How about I just cook something up and then you guys can vote on which is better?

**Comment by PullJosh** [comment]
Looking back on what I wrote, it's clear that you guys are taking the brunt of my perfectionism. I'll try to tone it down. :package:

**Comment by BookOwl** [comment]
> We have massive border-radiuses (radii?) and box-shadows

I personally like both of those.

> The navigation bar, on hover, is a terrible off-green

I agree with you on that.

> there is a bunch of unused space.

Again, I agree with you.

> Take a look at the site from the eyes of a potential user. Nothing about it appears professional. Even on a subconscious level, the site as it stands will turn away users.

I'm not sure about that. While the site could be better, I don't think it's _that_ bad.

> How about I just cook something up and then you guys can vote on which is better?

Sounds like a good plan to me.

**Comment by towerofnix** [comment]
I like the box shadows but maybe more transparent / less thick, same for the border-radiuses but maybe less round.

**Comment by PullJosh** [comment]
> I like the box shadows but maybe more transparent / less thick, same for the border-radiuses but maybe less round.

Agreed. I like very subtle doses of each (4-6px border radius, extremely small and subtle box shadow).

**Comment by matthewr6** [comment]
Agreed with the border-radius and box-shadow properties.

Not sure what color green we want/have been using for the nav bar and other parts of the site...  Preferably whatever's in the editor's bar

> Take a look at the site from the eyes of a potential user. Nothing about it appears professional. Even on a subconscious level, the site as it stands will turn away users.

Yeah we get that you don't like it :P 

> How about I just cook something up and then you guys can vote on which is better?

Ok, sounds fair

Why not just modify the current stuff though?  (also because our CSS **_must be**_ reused throughout pages - so much cleaner!)  It seems the current consensus is wider stuff, less box shadow/border radii, and slightly different navbar colors on hover

> my perfectionism

It's not as much perfectionism as "I prefer it better my way, so it has to be that way or else it won't work", imo (of course not regarding the box shadow and border radii)

**Comment by PullJosh** [comment]
> It's not as much perfectionism as "I prefer it better my way, so it has to be that way or else it won't work", imo (of course not regarding the box shadow and border radii)

Well... yeah. Whatever it is, I have a terrible case of it. Sorry for being a bit brutal. It's just the way I am. :package:

**Comment by matthewr6** [comment]
It's fine - I'm like that too a lot :P

**Comment by PullJosh** [comment]
> It's fine - I'm like that too a lot :P

You're allowed to do it for the backend if I can do it for the frontend. xD

**Comment by PullJosh** [comment]
In terms of UX... Are we okay with having a different homepage depending on whether or not the user is signed in, or do we want them identical?

**Comment by matthewr6** [comment]
Fairly identical would probably be best, though a big intro picture (not necessarily a picture tho - kinda like the thing in your mockup) inserted between the navbar and the actual content would be cool

**Comment by PullJosh** [comment]
So something like Scratch has?

**Comment by matthewr6** [comment]
Yeah
(I was talking about the picture/notreallyapicture being like that screenshot we have with the labels)

**Comment by PullJosh** [comment]
Sounds cool!

What are we looking at in terms of required browser support? (Read as: is [flexbox](http://caniuse.com/#search=flexbox) okay?)

**Comment by matthewr6** [comment]
No, read as: What's the oldest IE version we want to support?  :P 
I think IE 11.

**Comment by PullJosh** [comment]
Okay! I'll go with flexbox then. :)
(Should I be discussing on gitter?)

**Comment by towerofnix** [comment]
> No, read as: What's the oldest IE version we want to support? :P
> I think IE 11.

Edggggge pls :P

jk, IE 11 I guess.

**Comment by PullJosh** [comment]
Dang... I thought I was getting somewhere, but then I added some text and it ended up really hard on the eyes. :(

Determined to make this perfect.

**Comment by towerofnix** [comment]
Oh no, did you leave a spare

```
* {
  font-size: 80px !important;
  color: red !important;
}
```

somewhere? :P

(EDIT: more terrible)

**Comment by matthewr6** [comment]
xD

**Comment by PullJosh** [comment]
Does everyone agree with firedrake that a light colored site is better? (because imo dark looks _way_ better and matches the editor much more nicely)

Give this a "thumbs up" reaction for light and a "thumbs down" reaction for dark.

**Comment by towerofnix** [comment]
(Confused bc I'm not sure which to choose, your post was written well and stuff)

On the one hand I prefer light webpages but I'm often setting my text editor to dark colors..

**Comment by PullJosh** [comment]
I've come to a bit of a compromise, or at least the idea for one. We'll see how it goes.

**Comment by PullJosh** [comment]
@Firedrake969 How hard is it, from the backend side of things, to put the sign up/log in forms on the homepage? (I'm hoping to do something like scratch does.)

**Comment by matthewr6** [comment]
It'd probably be best to have just one of the forms (I'd prefer the login form)

**Comment by PullJosh** [comment]
Okay... so login is a little popup thing like Scratch, but sign up is a link to another page?

**Comment by BookOwl** [comment]
> Okay... so login is a little popup thing like Scratch, but sign up is a link to another page?

That's professional web design for you! xP

**Comment by PullJosh** [comment]
> That's professional web design for you! xP

It's a shame we can't do better than the big, monstrous companies. It doesn't seem like it would be that hard... (I'm no expert, but it seems like you could just have an input of the `hidden` variety which states whether the user is signing up or logging in. Is this not possible?)

**Comment by matthewr6** [comment]
It is actually somewhat difficult and, as importantly, breaks Django's design patterns.  Just have two separate pages for login/signup...

It is possible, but given that it does break the design patterns of Django and does require a potentially substantial amount of effort, maybe we should vote?

Thumbs up if you want to have them on different pages, thumbs down if not @ElementalCode/programmers .

**Comment by PullJosh** [comment]
Okay.

**Comment by PullJosh** [comment]
Running into readability problems again. Our green color really doesn't have any good color you can place on top of it and have be readable. :(

**Comment by matthewr6** [comment]
White?
What's the hex code?

**Comment by PullJosh** [comment]
All colors are listed here: https://github.com/ElementalCode/Elemental/wiki/Dev-Notes

Here's what the call to action button looks like currently:
![image](https://cloud.githubusercontent.com/assets/5458180/14034016/782a6998-f1f5-11e5-84e8-eecc30368ac6.png)
Clearly there's still work to be done.

**Comment by matthewr6** [comment]
(do remember that this isn't merely a cover page, please and thanks)

**Comment by PullJosh** [comment]
Yup! As I said, still lots of work to be done.

**Comment by BookOwl** [comment]
I can read it just fine... :confused:

**Comment by matthewr6** [comment]
I still think we should be improving our current design, not creating a completely new one, tbh

**Comment by PullJosh** [comment]
> I still think we should be improving our current design, not creating a completely new one.

What is the benefit?

**Comment by matthewr6** [comment]
Rather, what is the benefit of creating a new design in place of a functional design we currently have?  That is what is being proposed, and is what must be proved - not the other way around.
However, one reason is that there's already a base design to work from.
(I think you still have yet to give a concrete reason why the design is bad aside from points that we agree on...  If there are legitimate points, it'd be helpful to know instead of just doing stuff and then having you do things over because you want it differently)

**Comment by PullJosh** [comment]
> Rather, what is the benefit of creating a new design in place of a **functional** design we currently have?

I think that's where our viewpoints differ. To me, just because something is functional doesn't mean it's good. I'm a more artsy person, which is definitely not the standard programmer mindset.

To be fair, you're probably right that I should have started from what we already have. At this point, though, I'm too far in for it to be worth going back and starting over.

My plan was to create a new branch once everything is done and shift around all of the styles and things myself, and just have you guys check. I guess I'm not really sure why you're so adamant about removing my workload? Honestly, I find projects like this to be _super_ fun.

**Comment by towerofnix** [comment]
Rewrite all the things! :)

I do that [too much](https://www.youtube.com/watch?v=XcUUY8ziTuk).

**Comment by matthewr6** [comment]
> To me, just because something is functional doesn't mean it's good.

Exactly - it's currently functional, which is the first step in having something be good.  It works, though it may not look nice.  That's why you can work off a functional thing to add to/change the design!

> I guess I'm not really sure why you're so adamant about removing my workload?

That's certainly not my goal - I'm just trying to think about the most efficient way to do things.

I'm definitely OK with complete rewrites if that's the only way to do something - just look at my gopher-brain repo :P

**Comment by PolyEdge** [comment]
> The navigation bar, on hover, is a terrible off-green

The navbar is seafoam

**Comment by PolyEdge** [comment]
And also, **I** wrote the frontpage.

**Comment by matthewr6** [comment]
I think he meant on hover (which I still like tbh)

don't worry, not everyone completely dislikes the page :P

**Comment by PolyEdge** [comment]
Me too

**Comment by TheInitializer** [comment]
Am I the only one that agrees?

## 

> NO DARK BACKGROUNDS PLEASE

I like dark backgrounds... 
at least have one on the non-logged-in splash :package:

**Comment by PullJosh** [comment]
> at least have one on the non-logged-in splash :package:

That was my plan exactly.

**Comment by matthewr6** [comment]
continuity though...

**Comment by PullJosh** [comment]
> continuity though...

It's like a little banner.
Here's a beautiful diagram I made to show the general idea:
![image](https://cloud.githubusercontent.com/assets/5458180/14057280/7ee5de3e-f2d1-11e5-8dd1-f220f68005c3.png)
The black banner\* is only there for signed out users. People who are signed in just get strait to the project carousels (unless we want to stick something else in there like blog posts.)

[]()\* not actually black in real design

**Comment by matthewr6** [comment]
K, kinda looks like what we have now :P  so cool

**Comment by PullJosh** [comment]
In terms of making the site responsive... How small of screens should we support?

I wouldn't imagine mobile users are going to be a large portion of our audience. (If we really want to know, we can always use Google Analytics to find out the exact resolutions of users' devices.)

**Comment by towerofnix** [comment]
I'd say at least 200px wide screens should be supported. :P

**Comment by matthewr6** [comment]
Tablet.

**Comment by PullJosh** [comment]
> I'd say at least 200px wide screens should be supported. :P

I hope you're joking.

> Tablet.

That seems good.

**Comment by towerofnix** [comment]
> > I'd say at least 200px wide screens should be supported. :P
> 
> I hope you're joking.

I'm not joking about 200px for the actual website and previews of sites people have created with Elemental. Tablet size is good for the editor though.

**Comment by PullJosh** [comment]
> > > I'd say at least 200px wide screens should be supported. :P
> > 
> > I hope you're joking.
> 
> I'm not joking about 200px for the actual website and previews of sites people have created with Elemental. Tablet size is good for the editor though.

![image](https://cloud.githubusercontent.com/assets/5458180/14063014/7e791d4c-f38c-11e5-85d4-dd78186ed18e.png)

**Comment by towerofnix** [comment]
Oh wow haha. I meant 300px :P

**Comment by PullJosh** [comment]
> Oh wow haha. I meant 300px :P

A bit of googling seems to suggest that 320px is generally the lowest any sane person goes, as it's the size of older (smaller) iphones in portrait mode. I _highly_ doubt we'll be getting many phone visitors, so going lower than about 500px will be low priority.

**Comment by matthewr6** [comment]
We're not quite sane tho :P 
+1 for 500px

**Comment by PullJosh** [comment]
> We're not quite sane tho :P 

Shhh...

**Comment by PolyEdge** [comment]
I feel kind of bad now. Is it that darn `border-radius` or is it just terrible :stuck_out_tongue_winking_eye:

**Comment by PolyEdge** [comment]
:cry:

**Comment by PolyEdge** [comment]
I'm pointless then if you're going to overhaul the frontend

**Comment by PolyEdge** [comment]
Would longer boxes be better?
http://prntscr.com/akjuwc

**Comment by PolyEdge** [comment]
I'm depressed

**Comment by iamunknown2** [comment]
@Dylan5797 I think we should use something else other than bright orange...

It would also be nice if it was responsive and adapted to the size of the screen. Maybe something like this:

``` css
#theIdOfTheDivWhichContainsTheSignUpThingy {
    min-width: 50%;
    width: 300px;
    max-width: 80%;
}
```

**Comment by PullJosh** [comment]
> I'm depressed

O_O I hope you're exaggerating.

**Comment by PullJosh** [comment]
Okay, I'm going to share what I've got so far so that you guys can nitpick it. I want you to be as mean to mine as I was to Dylan's. Pick it apart and tear it to shreds. That way we can get it perfect.

**I currently have only finished the navigation bar and banner.** I'm aware that this homepage will need to be more than just an advertisement (also show "featured" pages and things), but I'm not to that point yet. Right now just take a look at the navigation bar and logged-out banner:

http://codepen.io/PullJosh/pen/YqVazv

Tear it apart, please!

(Also, big space at bottom is so I could test out scrolling on the page and the fancy little effect I added there.)

**Comment by PullJosh** [comment]
(Sorry for triple post...)

Here are my current concerns about the design:
1. Logged-out banner area might be too tall (opinions?)
2. the "or" between the two green buttons at the bottom looks kind of ugly
3. The links in the navigation bar don't really change at all on hover (only your cursor looks different). Is this bad for usability?
4. I tried to make everything nice and responsive, but the header text looks too small at a width of `851px` to `1000px`.
5. I am now less concerned about readability of the green buttons than I was at one point, but it's still on the back of my mind
6. There's a little shadow on the navigation bar which appears as you scroll (much like when you scroll down in a Google doc). This runs on `window.onscroll`. The concern is that this might be harsh on older computers.

**Comment by towerofnix** [comment]
![screen shot 2016-03-27 at 10 11 21 am](https://cloud.githubusercontent.com/assets/9948030/14065556/f2ab50b8-f404-11e5-81db-6ccffcbaf744.png)

> The concern is that this might be harsh on older computers.

I feel sad for the people who have computers like that. :package:

**Comment by TheInitializer** [comment]
I honestly love it. I like how we brought that purple back - tbh, coral is really ugly if overused. I'll try to critique it later I guess

**Comment by TheInitializer** [comment]
http://codepen.io/TheInitializer/pen/MyoVgY did a few little tweaks
- tooltip thingies are transparent when hovered over
- register and sign in both have transitions
- not much else :package:

**Comment by mrjacobbloom** [comment]
My biggest concern is that this new design is starting to feel like a landing page, and I don't think the logged-in version will look any good without the screenshot widget. You'll never see these scrolly-big-picture-minimalistic landing-pages double as a dashboard for a logged-in user, there's always a separate dashboard/homepage. The first example that comes to mind is Google Drive: only logged-out users get to see their big scrolly pictures. If we go forward with this design, we'll have to create a completely separate homepage for users

okay, other concerns:

> - tooltip thingies are transparent when hovered over

I feel like that should be the other way around, transparent until hovered-over. Or maybe even have most of them transparent and have them cycle which one is opaque every few seconds (like a slideshow). Or, better yet, have the text for the inactive tooltips collapse into an :information_source: symbol or something.

> strange box shadow

yeah that's gotta go

> 2: the "or" between the two green buttons at the bottom looks kind of ugly
> 3: The links in the navigation bar don't really change at all on hover (only your cursor looks different). Is this bad for usability?
> ...
> 6: This runs on window.onscroll. The concern is that this might be harsh on older computers.

2: I like the "or" as-is, or at least the idea behind it
3: it won't confuse anyone but I'd like it a lot better if they had a hover effect
6: yeah old computers hate that, but it's better than setInterval :package: enough other sites use it that I say leave it in. Besides, it looks kickbutt

Other thoughts/complaints (NOTE: on a screen that falls under `max-width: 1634px`):
- overall it's looking a lot better than the current site. Green and orange was a bad color combo
- the first tooltip sticks way too far off the left edge. Maybe shorten the text to "Share with one click" or "Share with friends"
- the tooltips look too big and childish. Try a lighter font-weight and less border-radius. Or just make them smaller
- I like the font of the big text. Is it the same one we use for blocks? It looks :+1: here
- I like the navbar shadow effect; however, once you can see its border it looks too tall next to the 500px-wide content. Maybe shrink the padding as you scroll
- the green buttons at the bottom need to respond to hover. They're also really close to the width of the screenshot and it bugs me that they don't lign up
- make the screenshot a link. Either to the editor or the register page.
- we need to decide on a logo and stick with it. In the navbar. It can be more low-key than what we currently have but it can't just be text. My suggestion is dark-gray angle brackets around the word "Elemental" that become light on hover

**Comment by PullJosh** [comment]
> My biggest concern is that this new design is starting to feel like a landing page.

I think you're right. One option would be to try to shrink down the landing image area. Let's wait for thoughts from others to see what happens.

> I like the "or" as-is, or at least the idea behind it

Okay! I might tweak how it looks just a bit, but it should stay functionally the same.

> (_about nav bar_:) it won't confuse anyone but I'd like it a lot better if they had a hover effect

I was thinking about making them green on hover, but I don't want them to look like the first ("Elemental") link.

> yeah old computers hate that, but it's better than setInterval :package: enough other sites use it that I say leave it in. Besides, it looks kickbutt

Yay! :D

> the first tooltip sticks way too far off the left edge. Maybe shorten the text to "Share with one click" or "Share with friends"

Very good point. I'll definitely fix that one.

> the tooltips look too big and childish. Try a lighter font-weight and less border-radius. Or just make them smaller

Interesting... I can totally see where you're coming from with the bold font being too large. I kind of like the border radius being so large (it makes the ends perfectly circular). I'll continue to play around with those. (Making font thinner could potentially fix the previous issue!)

> I like the font of the big text. Is it the same one we use for blocks? It looks :+1: here

Yup. Source sans pro. :)

> I like the navbar shadow effect; however, once you can see its border it looks too tall next to the 500px-wide content. Maybe shrink the padding as you scroll

I'm not sure I agree on this one. Anybody else have opinions?

> the green buttons at the bottom need to respond to hover. They're also really close to the width of the screenshot and it bugs me that they don't lign up

Agreed. The width thing comes down to circumstance (buttons are static size, image scales). I'll try to get them all lined up nice with a hover effect.

> make the screenshot a link. Either to the editor or the register page.

I'll think about it. If it becomes a link, it should definitely be to the editor.

> we need to decide on a logo and stick with it. In the navbar. It can be more low-key than what we currently have but it can't just be text. My suggestion is dark-gray angle brackets around the word "Elemental" that become light on hover

I'll give that a try.

**Comment by PullJosh** [comment]
Is this version of the "or" better or worse?

![image](https://cloud.githubusercontent.com/assets/5458180/14076947/66adb6b8-f4b4-11e5-81a4-531c12d61c9c.png)

**Comment by iamunknown2** [comment]
@PullJosh +1 because it makes it more apparent to the user that an account is required to save to the cloud, without having to sacrifice much aesthetically.

However, it might be better if the blocks were on different lines, with the "or" in between them.

**Comment by PullJosh** [comment]
> However, it might be better if the blocks were on different lines, with the "or" in between them.

Try shrinking down your browser (or the codepen preview window). ;)

**Comment by mrjacobbloom** [comment]
I like the "or" better without the horizontal lines

**Comment by PolyEdge** [comment]
I like all this, but what about every other page on the website...

**Comment by PolyEdge** [comment]
Oh and I was joking in all my posts further up...

**Comment by mrjacobbloom** [comment]
> I like all this, but what about every other page on the website...

shhhh don't think so hard

> Oh and I was joking in all my posts further up...

thank goodness, I was actually woried :package:

**Comment by PolyEdge** [comment]
lol

**Comment by PolyEdge** [comment]
oh did i make the new slogan "Learn to develop on the web with an easy drag-and-drop interface."

**Comment by PolyEdge** [comment]
I'm assuming that we should have a separate css file for the front page, right?

**Comment by PolyEdge** [comment]
And also, is the text going over top of the carousel in the commented out code intentional? I think it's a bit weird tbh.

**Comment by mrjacobbloom** [comment]
> oh did i make the new slogan "Learn to develop on the web with an easy drag-and-drop interface."

can we change it to "Learn to develop websites with an easy drag-and-drop interface."?

> And also, is the text going over top of the carousel in the commented out code intentional? I think it's a bit weird tbh.

lol I think that's why it's commented out :stuck_out_tongue_closed_eyes:

**Comment by TheInitializer** [comment]
> Or, better yet, have the text for the inactive tooltips collapse into an :information_source: symbol or something.

[Done.](http://codepen.io/TheInitializer/pen/MyoVgY/)

**Comment by TheInitializer** [comment]
(please improve on my terrible blue dots)
(any ideas for colours to use for the dots?)
(lol I feel like I'm typing in a Lisp-like language)

**Comment by TheInitializer** [comment]
Also I like the pocket shadow thingy

**Comment by mrjacobbloom** [comment]
> (please improve on my terrible blue dots)
> (any ideas for colours to use for the dots?)
> (lol I feel like I'm typing in a Lisp-like language)

`please(improve, on, my, terrible, blue, dots);`
I was thinking the tooltip text collapses, but that works too :P those blue dots are actually terrible tho haha

**Comment by TheInitializer** [comment]
Lolyeah. We need a colour that looks good and stands out from the background (or maybe just bigger dots :package: )

**Comment by TheInitializer** [comment]
Here's an example of something like this: http://komodoide.com/

**Comment by PolyEdge** [comment]
> (please improve on my terrible blue dots)

Make the circles jump every once in a while to give the idea that they're interactive.

> (any ideas for colours to use for the dots?)

RRRRAAAAIIIINNNNBBBBOOOWWWW
![lel bad reference](https://s3.amazonaws.com/colorslive/png/102387-j7xEX6rPF99qdTjC.png)

> (lol I feel like I'm typing in a Lisp-like language)

+9000

> Here's an example of something like this: http://komodoide.com/

I'll might be able to do it if I had time.

**Comment by mrjacobbloom** [comment]
> Here's an example of something like this: http://komodoide.com/

I like the radar animation. you can probably pull that off with CSS alone if you use animations and the outline property.

**Comment by TheInitializer** [comment]
http://codepen.io/TheInitializer/pen/MyoVgY
Better colour, box-shadow, semi-transparent. I'll look into a radary thing

**Comment by TheInitializer** [comment]
So I made a radary thing but for some reason it doesn't show the popup when I hover over it (but it does without the radary thing). Probably because I couldn't figure out how to use ::after so I just copied and pasted the div :package: Anyway I'll leave you to debug it: http://codepen.io/TheInitializer/pen/MyoVgY

**Comment by PullJosh** [comment]
Any ideas on what the UX benefits are for the dots?

> And also, is the text going over top of the carousel in the commented out code intentional? I think it's a bit weird tbh.

Things always look awful when they're a work in progress. That's why it was commented out. :stuck_out_tongue_winking_eye:

**Comment by mrjacobbloom** [comment]
Fixed the dots: http://codepen.io/anon/pen/EKvYpa

> Any ideas on what the UX benefits are for the dots?

_shrug_ I'm still rooting for a slideshow-type-thing where only one tooltip is visible/noticeable at a time, so we need a way to hide the other hints while still making them accessible.

**Comment by mrjacobbloom** [comment]
Ohh the pocket wasn't registering as a pocket for me. I fiddled with the CSS until it did, try this:
`box-shadow: 0 -17px 20px -13px rgba(68,76,84,0.9), 0 -5px 4px -5px rgba(0,0,0,0.5);`

http://codepen.io/anon/pen/vGJEBb (I was also playing with the dots on hover)

**Comment by TheInitializer** [comment]
> Fixed the dots

I already fixed them, just forgot to post here :package: 
Now I'm trying to get the animation to stop when one of the dots is hovered over :page_facing_up: 

> Any ideas on what the UX benefits are for the dots?

The tooltips don't cover the image, so you can see more

**Comment by rumanti** [comment]
Nooo, I design the pages! _cries_ - Actually, just kidding - and I don't design the navigation bar :package: 

Personally, I'm not happy with the childishness of the design - I _thought_ that the design was supposed to be child-friendly, probably because of the "cveryone learn computers now!" trend, and I was like, "ugh, you can't just abort just because you don't like the idea! Get on with it!" I am pretty famous in life for almost never getting anything done because of my stress for perfectionism, so I dropped my standard ... So that I get something done. 

About the excess border-radius/radii, I just hate sharp corners for no reason. And box-shadows - erm, I'm not coming up with any lame excuse because it does look horrible. Why no dark background, though? I prefer dark honestly because it looks cool (the temperature) for my eyes. My eyes are definitely not perfect - I wear glasses. Sometimes.

---

My conclusion:
I got the target audience messed up. 
I'm not really sure what the team actually want. Professional? Portofolio style? Material design? Flat?
I was emotionless and didn't try thinking deeply about the potential users' opinion.

---

My ideas:
Discuss an "official" color palette (Scratch is blue and orange; GitHub is white and deep lime green; etc) or at least theme (monochrome, for example)
Give the navigation bar a neutral color (white, gray, black)
Decide the layout style (compact, for example)
Decide the design style (flat, for example)
Decide some basic rules (no bright colors, for example)

There :)

**Comment by PullJosh** [comment]
> Discuss an "official" color palette (Scratch is blue and orange; GitHub is white and deep lime green; etc)

https://github.com/ElementalCode/Elemental/wiki/Dev-Notes#ui---user-interface

> Give the navigation bar a neutral color (white, gray, black)

Any reason?

> Decide the layout style (compact, for example)
> Decide the design style (flat, for example)

I'm thinking flat w/ subtle shadows and relatively spacious.

---

I haven't been working on this lately because I'm still on vacation. I'll start working on it again when I get back. :)

**Comment by quat1024** [comment]
Hi guys!

> About the excess border-radius/radii, I just hate sharp corners for no reason. And box-shadows - erm, I'm not coming up with any lame excuse because it does look horrible.

A really nice compromise is to use a 3px border-radius. Box shadows also look great when used for a **subtle** effect (just slightly darker than the background and about 3 pixels long, no blurring).

That type of effect is what Google uses for Cards UI/Google Now stuff - seriously, [look at them](http://cdn.androidcommunity.com/wp-content/uploads/2015/01/Google-Now-Launcher.png), they know what they're doing.

Sharp corners and flat-as-a-pancake design is boring, but then again, `border-radius: 1000px;` might have been cool in 2008. You have to find a medium. That said, whats _"in"_ in design, changes just as often as what's _"in"_ in fashion. It really depends on your audience

**Comment by PullJosh** [comment]
It might be bad that I only just thought of this, but... Do we have a way to get screenshots of each page like CodePen does?

**Comment by matthewr6** [comment]
Yeah it's already implemented

**Comment by PullJosh** [comment]
> Yeah it's already implemented

Wow! You're on top of things. Is a square image okay?

**Comment by matthewr6** [comment]
It's 360x480 (or vice versa, can't remember) but it's pretty easy to change

**Comment by PullJosh** [comment]
> It's 360x480 (or vice versa, can't remember) but it's pretty easy to change

Most likely 480x360. Whatever the case, it shouldn't be too hard to change things around in the design. :)

**Comment by matthewr6** [comment]
I can work w/pretty much whatever size

**Comment by rumanti** [comment]
> https://github.com/ElementalCode/Elemental/wiki/Dev-Notes#ui---user-interface

But it looks like light backgrounds are winning!

> Any reason? (Give the navigation bar a neutral color (white, gray, black))

Bright navigation bars demands too much attention.

> Tear it up, please!
1. I don't like the fact that the background and the navigation have the same background color, with only box-shadow to separate them. The navigation bar should be a little darker (try #343C44) in my opinion.
2. The Register button seems too sharp-ish and doesn't  fit with the rest of the "fluid" design. I vote 8px box-radius.

Other than that, I like the pocket shadow.

**Comment by PullJosh** [comment]
> But it looks like light backgrounds are winning!

Yeah, I just grabbed a few light grays (white and #ddd, iirc)

> Bright navigation bars demands too much attention.

That's fine! But I think we can still have some color to it.

> I don't like the fact that the background and the navigation have the same background color, with only box-shadow to separate them. The navigation bar should be a little darker (try #343C44) in my opinion.

I'm not feeling it.
![image](https://cloud.githubusercontent.com/assets/5458180/14424620/1544fdde-ffb0-11e5-9587-26c93180a26a.png)

> The Register button seems too sharp-ish and doesn't fit with the rest of the "fluid" design. I vote 8px box-radius.

Again, I'm not sure I'm a fan.
![image](https://cloud.githubusercontent.com/assets/5458180/14424669/60fd3e76-ffb0-11e5-9d7f-6fefafcba0b0.png)
You say it fits in better, but most things on the page have a 4px border-radius.

**Comment by towerofnix** [comment]
I vote.. 6px `border-radius`? :P

I can't stand when the navigation bar is the exact same color as the main background. It makes it hard to distinguish between the nav and the content. So I'd go for something a bit lighter (but not _much_ lighter) than the nav in the demo above.

Just my opinion..

**Comment by PullJosh** [comment]
> I can't stand when the navigation bar is the exact same color as the main background.

Bear in mind that the showcase image will only appear when the user is signed out. :)

**Comment by iamunknown2** [comment]
Why doesn't @PullJosh fork master and put changes and issues/suggestions there instead of one big thread?

**Comment by matthewr6** [comment]
Issues are for discussion, but branches aren't.

`4px` border-radius is what bootstrap uses, if that matters...

**Comment by iamunknown2** [comment]
The problem here is that ALL the issues are put in one big thread.

It would be much better if we had a dedicated branch to overhauling the UI.

**Comment by towerofnix** [comment]
afaik issues are linked to the repository, not branches. You'd need a new repo for a new set of issues. Or you could have a label saying something along the lines of "new design" or whatever that all website-is-ugly issues/pull requests would get.

**Comment by matthewr6** [comment]
+1 for labeling issues, since that's what labels are for

**Comment by iamunknown2** [comment]
Whoops I forgot :package: but it's better to keep the changes on a separate branch, rather than on a CodePen.

**Comment by matthewr6** [comment]
👍🏽

**Comment by PullJosh** [comment]
> It's better to keep the changes on a separate branch, rather than on a CodePen.

You're probably right. I'm using Codepen because I'm lazy. How about I create a branch once we have a first full homepage draft?

> 👍🏽

You do realize there's literally a button for that now... :package:

**Comment by matthewr6** [comment]
Commits don't have to be for complete things...  It'd be nice to be able to see how far you are in the same place as everything else

Not on mobile :P

**Comment by PullJosh** [comment]
> Not on mobile :P

MOBILE!? O_o

**Comment by matthewr6** [comment]
I meant when I use my iPhone to go on github issues xD

**Comment by PullJosh** [comment]
> I meant when I use my iPhone to go on github issues xD

I feel like somehow using a phone to access github is morally wrong.

**Comment by iamunknown2** [comment]
> I feel like somehow using a phone to access github is morally wrong.

PC master race :package:

**Comment by mrjacobbloom** [comment]
> > I feel like somehow using a phone to access github is morally wrong.
> 
> PC master race :package:

:iphone: :iphone: :iphone: :phone: :iphone: :iphone:

**Comment by iamunknown2** [comment]
Wait... You can only assign one person at a time!??

**Comment by mrjacobbloom** [comment]
Before I saw that it was an accident I was like "ohhhh shade thrown"

Sent from my iPhone.

**Comment by matthewr6** [comment]
@iamunknown2 yeah

**Comment by PullJosh** [comment]
As tends to happen when I work on things, I've hit a burnout point where I don't really have much more energy to give on this one.

I took a break once I got to here, and didn't really get going again:
http://codepen.io/PullJosh/pen/PNjQmO

If someone else wants to take over that'd be great.
<sup><sup>I'm terrible</sup></sup>

**Comment by matthewr6** [comment]
> someone else takes over

a while later...

> let's redo it again

I like it - only thing I'd suggest is some sort of text/background color change for the navbar when you hover over create/explore/sign in

**Comment by BookOwl** [comment]
Me too. The only thing that I would change is to center the footer links.

**Comment by iamunknown2** [comment]
Uh oh, it looks like Elemental is about to go dormant...

Anyway, can anybody fork master and push?

**Comment by TheInitializer** [comment]
I don't like the T-shaped things around the "or" and the big green buttons. I liked it before

**Comment by matthewr6** [comment]
I don't mind the green buttons but I don't like the T things and the "or" should be bigger, IMO

**Comment by TheInitializer** [comment]
I meant just the T things, not the buttons. Lol

**Comment by TheInitializer** [comment]
from a comment in the css:

> Is it possible to only serve these when the user is logged out?

yes

**Comment by TheInitializer** [comment]
here's my whole thing, based off of my original fork, nickynouse's fork of my fork and pulljosh's new thing. It's basically all the stuff I like :package: 
http://codepen.io/TheInitializer/pen/wGxEYJ?editors=1100

one thing:

when one of the dots is hovered, it stops, but the others keep on going. Is there any way to stop all of the dots when one is hovered? That way they won't get all out of sync

also, do people like the hover effect for the dots? I wasn't so sure about it

**Comment by mrjacobbloom** [comment]
> when one of the dots is hovered, it stops, but the others keep on going. Is there any way to stop all of the dots when one is hovered? That way they won't get all out of sync

I don't have free time to try this, but you could try putting the dots inside a 0x0 box, that way the parent will only have a hover effect when one of them is hovered-over

**Comment by matthewr6** [comment]
@ElementalCode/programmers could someone implement this?  I'm working on a featured projects system but it depends on the front page designs

also clean up the CSS so that way we only need one file for the whole site instead of a single file for every page :/  Right now it's a mess, tbh - this is why everyone should run/develop locally instead of using codepen

**Comment by TheInitializer** [comment]
https://scratch.mit.edu/discuss/topic/134554/?page=53#post-1977743 proof :package:

**Comment by matthewr6** [comment]
site design != editor design
weren't the colors supposed to be temporary but we've never changed them?  :P

**Comment by towerofnix** [comment]
>  weren't the colors supposed to be temporary but we've never changed them? :P

_changes a couple variables_
:tada:

**Comment by mrjacobbloom** [comment]
are we still waiting for someone to commit this? I can commit this

**Comment by matthewr6** [comment]
Sure

**Comment by mrjacobbloom** [comment]
Hey random question: The signup page keeps giving me an error that Localhost isn't approved for captcha. Is there a way around that so I can test the logged-in stuff?

**Comment by mrjacobbloom** [comment]
Oh wait that's totally what createsuperuser does. I'm dumb, nvm
